tests/WhoopsMiddlewareTest.php

```diff
@@ -4,49 +4,76 @@
 
 use Exception;
 use Franzl\Middleware\Whoops\WhoopsMiddleware;
+use Middlewares\Utils\Factory;
+use Middlewares\Utils\FactoryInterface;
 use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseFactoryInterface;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\StreamInterface;
 use Psr\Http\Server\RequestHandlerInterface;
-use Zend\Diactoros\Response\TextResponse;
-use Zend\Diactoros\ServerRequest;
 
 class WhoopsMiddlewareTest extends TestCase
 {
+    public function setUpForError($expectedContentType)
+    {
+        $responseMock = $this->createMock(ResponseInterface::class);
+        $responseMock->expects($this->once())->method('withBody')->willReturnSelf();
+        $responseMock->expects($this->once())
+            ->method('withHeader')
+            ->with('Content-Type', $expectedContentType)
+            ->willReturnSelf();
+        $responseFactoryMock = $this->createMock(ResponseFactoryInterface::class);
+        $responseFactoryMock->expects($this->once())->method('createResponse')->with(500)->willReturn($responseMock);
+
+        $streamMock = $this->createMock(StreamInterface::class);
+        $streamFactoryMock = $this->createMock(StreamFactoryInterface::class);
+        $streamFactoryMock->expects($this->once())->method('createStream')->willReturn($streamMock);
+
+        $factoryMock = $this->createMock(FactoryInterface::class);
+        $factoryMock->expects($this->once())->method('getResponseFactory')->willReturn($responseFactoryMock);
+        $factoryMock->expects($this->once())->method('getStreamFactory')->willReturn($streamFactoryMock);
+
+        /** @var FactoryInterface $factoryMock */
+        Factory::setFactory($factoryMock);
```

Comment:
I don't like this level of mocking - IMHO, it couples us too much to the actual implementation.

May I ask you to move Diactoros to `require-dev` and use it here? (Which would probably mean this file could remain unchanged.)

Reply:
Yes it's possible to revert this i guess and put Diactoros in `require-dev`. However the expectations in `setUpForError` are pretty explicit and equivalent in the mocks though, and reflect exactly the original tests without the dependency.

Reply:
Thanks for reverting! :hugs:  I really do think it serves the test better not to mirror the implementation that closely.